pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -222,13 +222,19 @@ static PyObject *get_values(PyObject *obj) {
 
     PRINTMARK();
 
-    if (PyObject_HasAttrString(obj, "_internal_get_values")) {
+    if (PyObject_TypeCheck(obj, cls_index) || PyObject_TypeCheck(obj, cls_series)) {
```

Comment:
I guess will see on test cases, but is this really required? AFAICT on master `_internal_get_values` and `values` return the same for a dt64tz array:

```python
>>> dts = pd.date_range(start="2020-01-01", freq="D", periods=3, tz="US/Pacific")
>>> dts._internal_get_values()
array(['2020-01-01T08:00:00.000000000', '2020-01-02T08:00:00.000000000',
       '2020-01-03T08:00:00.000000000'], dtype='datetime64[ns]')
>>> dts.values
array(['2020-01-01T08:00:00.000000000', '2020-01-02T08:00:00.000000000',
       '2020-01-03T08:00:00.000000000'], dtype='datetime64[ns]')
```

Reply:
its Categorical[dt64tz] and Sparse that make the extra `__array__` call necessary